examples/ex34p.cpp

```diff
@@ -0,0 +1,574 @@
+//                       MFEM Example 34 - Parallel Version
+//
+// Compile with: make ex34p
+//
+// Sample runs:  mpirun -np 4 ex34p -o 2
+//               mpirun -np 4 ex34p -o 2 -pa
```

Comment:
I didn't think of that. I'll need to devise a modified problem to run on a mesh with tensor product elements... I think I can do that without too much effort.

Reply:
I'm seeing an issue on processor boundaries of a hexahedral mesh. I didn't see any such issue on mixed meshes so the problem may be limited to shared quadrilateral faces within the `ParSubMesh`. I may need a couple days to get back to this and dig deeper.

Reply:
I believe this is now fixed. I've also added support for a hex-only mesh in both examples so that we can test the partial assembly options.

I've very glad you caught this, @psocratis. Thanks!